File: javascript_cypress/cypress/pages/GooglePage.js
class GooglePage {
    constructor() {}

    check_title(title) {
        cy.allure().startStep('Проверка названия страницы').endStep()
        cy.title().should('include', title)
    }

    open() {
        cy.allure().startStep('Открытие страницы').endStep()
        cy.visit('/');
    }

    fill_search_form(text) {
        cy.allure().startStep('Заполнения поля поиска значением {value}').endStep()
        cy.xpath("//input[@title='Поиск']").type(text)
    }

    click_button() {
        cy.allure().startStep('Клик по кнопке поиска').endStep()
        cy.xpath("//input[@title='Поиск']").type('{enter}')
    }

    click_button_with_text(text) {
        cy.allure().startStep('Клик по кнопке с текстом - {text}').endStep()
        cy.xpath(`//td/div/div[text() = "${text}"]`).click()
    }

    check_result(expression, result) {
        cy.allure().startStep('Проверка результата').endStep()
        cy.xpath('//span[@jsname="ubtiRe"]').should('have.text', expression)
        cy.xpath('//span[@jsname="VssY5c"]').should('have.text', result)
    }
}

export default new GooglePage();

File: csharp_nunit/csharp_nunit/pages/GooglePage.cs
using NSelene;
using NUnit.Framework;

namespace csharp_nunit.pages
{
    public static class GooglePage
    {
        internal static void Open(string url = "/")
        {
            Selene.Open(url);
        }

        internal static void FillSearchForm(string value)
        {
            Selene.S("//input[@title='Поиск']").SetValue(value);
        }

        internal static void ClickButton()
        {
            Selene.S("//input[@title='Поиск']").PressEnter();
        }

        internal static void CheckTitle(string pageTitle)
        {
            var title = Selene.GetWebDriver().Title;
            StringAssert.Contains(pageTitle, title, "Заголовок страницы: " + title + ", ожидаемый результат: " + pageTitle);
        }

        internal static void ClickButtonWithText(string value)
        {
            Selene.S("//td/div/div[text() = '" + value + "']").PressEnter();
        }

        internal static void CheckResult(string expression, string result)
        {
            Selene.S("//span[@jsname='ubtiRe']").Should(Have.Text(expression));
            Selene.S("//span[@jsname='VssY5c']").Should(Have.Text(result));
        }
    }
}